### PR TITLE
Implement show and index endpoint for book

### PR DESCRIPTION
**Pull request by sebasdeldi**
## Summary
- Implements show and index endpoints for model
- Adds pagination
- adds AMS serializer

## Trello Card
https://trello.com/c/Bnh0AC32/21-crear-rutas-index-y-show-para-book

### CODE REVIEW

File: app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
```
Comment (nicozar95) [inline app/controllers/api/v1/books_controller.rb]: Make it inherit from `ApiController` once you add that controller in the other PR

File: app/controllers/api/v1/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      include Wor::Paginate
+      before_action :authenticate_user!
```
Comment (nicozar95) [inline app/controllers/api/v1/books_controller.rb:4]: Also move the:
```
include Wor::Paginate
```

File: app/controllers/api/v1/books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+      def index
+        render_paginated Book.all, each_serializer: BooksListSerializer
```
Comment (nicozar95) [inline app/controllers/api/v1/books_controller.rb]: You should not specify the serializer

File: app/serializers/api/v1/book_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class BookSerializer < ActiveModel::Serializer
+      attributes :id, :author, :title, :genre, :editor, :year, :image, :rents
+
+      def rents
```
Comment (nicozar95) [inline app/serializers/api/v1/book_serializer.rb]: You should not list the rents in the book

File: app/serializers/api/v1/books_list_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module Api
+  module V1
+    class BooksListSerializer < ActiveModel::Serializer
```
Comment (nicozar95) [inline app/serializers/api/v1/books_list_serializer.rb]: This class is unnecesary

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,12 @@
 Rails.application.routes.draw do
+  
+  namespace :api do
+    namespace :v1 do
+      get 'books', to: 'books#index'
```
Comment (nicozar95) [inline config/routes.rb]: A better way would be to use:
```
resources :books, only: [:index, :show]
```

Instead of specifying each get

**Review by nicozar95 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/books_controller.rb (hunk 3 above)
Comment (sebasdeldi) [reply app/controllers/api/v1/books_controller.rb]: According to the Apiari documentation, the response of both endpoints (index and show), return different forms of the Book object hence the need of specifying the serializer.

File: app/serializers/api/v1/books_list_serializer.rb (hunk 5 above)
Comment (sebasdeldi) [reply app/serializers/api/v1/books_list_serializer.rb]: I don't think it is
> According to the Apiari documentation, the response of both endpoints (index and show), return different forms of the Book object hence the need of specifying the serializer.

File: app/controllers/api/v1/books_controller.rb (hunk 3 above)
Comment (nicozar95) [reply app/controllers/api/v1/books_controller.rb]: The Apiari documentation has been deprecated in concern to this new training we are doing 

image

Use the same serializer to show and index in this case an as we go on in the training we could decide which is the best approach to the diferents controllers

**Review by nicozar95 — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 7]
```diff
@@ -28,6 +28,12 @@ gem 'puma', '~> 3.7'
 # Multi-client and secure token-based authentication
 gem 'devise_token_auth', '~> 1.0.0'
 
+# Back end json format
+gem 'active_model_serializers'
```
Comment (Gmarn) [inline Gemfile:32]: Lock the gem please ;)

File: Gemfile [hunk 8]
```diff
@@ -28,6 +28,12 @@ gem 'puma', '~> 3.7'
...
+
+# Pagination
+gem 'wor-paginate'
```
Comment (Gmarn) [inline Gemfile:35]: Same

**Pull request merged**